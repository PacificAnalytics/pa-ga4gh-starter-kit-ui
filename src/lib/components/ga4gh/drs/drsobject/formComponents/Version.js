import React from 'react';
import { SimpleTextField } from '../../../../common/form';

const Version = props => {
    return (
        <SimpleTextField
            id="version"
            label="Version"
            name="version"
            value={props.version}
            readOnly={props.readOnly}
            changeFunction={props.setVersion}
            helperText='Current version of the DRS Object, it should be updated each time the DRS Object is modified.'
        />
    )
}

export default Version;