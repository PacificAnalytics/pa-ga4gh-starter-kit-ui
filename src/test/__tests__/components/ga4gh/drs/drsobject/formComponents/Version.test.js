import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import {mockUpdateScalar} from '../../../../../../resources/MockFunctions';
import Version from '../../../../../../../lib/components/ga4gh/drs/drsobject/formComponents/Version';

afterEach(() => {
    mockUpdateScalar.mockClear();
})

test('SHOW <Version /> should handle populated field', () => {
    let {container} = render(
        <Version readOnly={true} version='1.0.0' setVersion={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Version')).toBeInTheDocument();
    let versionField = screen.getByRole('textbox');
    expect(versionField).toHaveValue('1.0.0');

    // version field is read only and cannot be edited
    expect(versionField).toHaveAttribute('readonly');
    userEvent.type(versionField, 'test value');
    expect(mockUpdateScalar.mock.calls.length).toBe(0);
});

test('NEW and EDIT <Version /> should handle empty field', () => {
    let {container} = render(
        <Version readOnly={false} version='' setVersion={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Version')).toBeInTheDocument();
    let versionField = screen.getByRole('textbox');
    expect(versionField).toHaveValue('');
    userEvent.type(versionField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});

test('NEW and EDIT <Version /> should handle populated field', () => {
    let {container} = render(
        <Version readOnly={false} version='1.0.0' setVersion={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Version')).toBeInTheDocument();
    let versionField = screen.getByRole('textbox');
    expect(versionField).toHaveValue('1.0.0');
    userEvent.type(versionField, 'test value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});