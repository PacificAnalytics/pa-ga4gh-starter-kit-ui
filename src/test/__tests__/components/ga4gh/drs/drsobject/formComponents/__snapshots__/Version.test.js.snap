// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <Version /> should handle empty field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="version"
      id="version-label"
    >
      Version
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="version-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="version"
        name="version"
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root"
      id="version-helper-text"
    >
      Current version of the DRS Object, it should be updated each time the DRS Object is modified.
    </p>
  </div>
</div>
`;

exports[`NEW and EDIT <Version /> should handle populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="version"
      id="version-label"
    >
      Version
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="version-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="version"
        name="version"
        type="text"
        value="1.0.0"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="version-helper-text"
    >
      Current version of the DRS Object, it should be updated each time the DRS Object is modified.
    </p>
  </div>
</div>
`;

exports[`SHOW <Version /> should handle populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="version"
      id="version-label"
    >
      Version
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="version-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="version"
        name="version"
        readonly=""
        type="text"
        value="1.0.0"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="version-helper-text"
    >
      Current version of the DRS Object, it should be updated each time the DRS Object is modified.
    </p>
  </div>
</div>
`;
